esp/platform/txsummary.cpp

```diff
@@ -21,10 +21,16 @@
 #include <algorithm>
 
 using std::find_if;
+using std::for_each;
 
 #define VALIDATE_KEY(k) if (!(k) || !(*k)) return false
 #define MATCH_KEY       [&](const Entry& entry) { return stricmp(entry.key.str(), key) == 0; }
 
+bool operator < (const StringBuffer& a, const StringBuffer& b)
```

Comment:
std::map is an ordered collection, with the default comparison implemented by std::less. std::less uses operator < for its comparisons. To use StringBuffer as a key in std::map requires a comparison for less than and no other comparison.